### PR TITLE
test: should reload sns proposal

### PR DESCRIPTION
**Pull request by mstrasinskis**
# Motivation

Cover the no-ballots-fix ([the pr](https://github.com/dfinity/nns-dapp/pull/4420)) with a test.

Steps done:
1. undo the fix
2. describe the correct behaviour in the test
3. test that it fail
4. restore the fix

# Changes

- test++

# Tests

Yes

# Todos

- [ ] Add entry to changelog (if necessary).
not necessary.

### CODE REVIEW

File: frontend/src/tests/lib/pages/SnsProposalDetail.spec.ts [hunk 1]
```diff
@@ -352,4 +353,92 @@ describe("SnsProposalDetail", () => {
       expect(await po.hasVotingToolbar()).toBe(true);
     });
   });
+
+  describe("An issue when the proposal w/o ballots from the store (from `proposalList` response) is used on proposal detail page", () => {
+    beforeEach(() => {
+      vi.clearAllMocks();
+
+      authStore.setForTesting(mockIdentity);
```
Comment (dskloetd) [inline frontend/src/tests/lib/pages/SnsProposalDetail.spec.ts:361]: We usually do this by calling [resetIdentity](https://github.com/dfinity/nns-dapp/blob/0c2ad8003baf3b15a9f56eecf84f09b0b5aae609/frontend/src/tests/mocks/auth.store.mock.ts#L27).

File: frontend/src/tests/lib/pages/SnsProposalDetail.spec.ts [hunk 2]
```diff
@@ -352,4 +353,92 @@ describe("SnsProposalDetail", () => {
...
+            ]),
+          },
+        ],
+      });
+
+      fakeSnsGovernanceApi.addProposalWith({
+        identity: mockIdentity,
+        rootCanisterId,
+        ...proposal,
+        proposal_creation_timestamp_seconds: proposalCreatedTimestamp,
```
Comment (dskloetd) [inline frontend/src/tests/lib/pages/SnsProposalDetail.spec.ts:412]: Why does the proposal in the store not have the same `proposal_creation_timestamp_seconds`?

**Review by dskloetd — COMMENTED** [review]
Thanks!

File: frontend/src/tests/lib/pages/SnsProposalDetail.spec.ts (hunk 1 above)
Comment (mstrasinskis) [reply frontend/src/tests/lib/pages/SnsProposalDetail.spec.ts]: Thanks, replaced.

File: frontend/src/tests/lib/pages/SnsProposalDetail.spec.ts (hunk 2 above)
Comment (mstrasinskis) [reply frontend/src/tests/lib/pages/SnsProposalDetail.spec.ts]: Good catch, it was just copy/pasted. The custom time was removed.

**Review by dskloetd — APPROVED** [review]
Thanks!

**Pull request merged**